one-shot-device/src/devices/device-registration.service.ts

```diff
@@ -0,0 +1,61 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { Model } from 'mongoose';
+import { defaultConfig } from '../configuration/configuration';
+import { InjectModel } from '@nestjs/mongoose';
+import { CreateDeviceRegistrationDto } from './dto/create-device-registration.dto';
+import { DeviceRegistrationDocument, DeviceRegistration } from './schemas/device-registration.schema';
+import { v4 as uuidv4 } from 'uuid';
+import { IdentityClient, ChannelClient } from 'iota-is-sdk';
+
+@Injectable()
+export class DeviceRegistrationService {
+	constructor(
+		@InjectModel(DeviceRegistration.name)
+		private readonly deviceRegistrationModel: Model<DeviceRegistrationDocument>
+	) {}
+
+	private readonly logger: Logger = new Logger(DeviceRegistrationService.name);
+
+	async createChannelAndIdentity() {
+		const channelClient = new ChannelClient(defaultConfig);
+		const identityClient = new IdentityClient(defaultConfig);
+
+		// create device identity
+		const deviceIdentity = await identityClient.create('my-device' + Math.ceil(Math.random() * 1000));
+
+		if (deviceIdentity == null) {
+			this.logger.error('Failed to create identity for your device');
```

Comment:
Ok here we cant just log the error since our program is not able to continue execution. So you need to throw an exception like done here:
https://docs.nestjs.com/exception-filters

as for instance
```
  throw new HttpException('Could not create the device identity.', HttpStatus.INTERNAL_SERVER_ERROR);

```

Reply:
Throw exception added.